open-api/rest-catalog-open-api.yaml

```diff
@@ -2597,12 +2598,15 @@ components:
           assert-last-assigned-partition-id: '#/components/schemas/AssertLastAssignedPartitionId'
           assert-default-spec-id: '#/components/schemas/AssertDefaultSpecId'
           assert-default-sort-order-id: '#/components/schemas/AssertDefaultSortOrderId'
-      type: object
-      required:
-        - type
-      properties:
-        type:
-          type: "string"
+      oneOf:
+        - $ref: '#/components/schemas/AssertCreate'
+        - $ref: '#/components/schemas/AssertTableUUID'
+        - $ref: '#/components/schemas/AssertRefSnapshotId'
+        - $ref: '#/components/schemas/AssertLastAssignedFieldId'
+        - $ref: '#/components/schemas/AssertCurrentSchemaId'
+        - $ref: '#/components/schemas/AssertLastAssignedPartitionId'
+        - $ref: '#/components/schemas/AssertDefaultSpecId'
+        - $ref: '#/components/schemas/AssertDefaultSortOrderId'
 
     AssertCreate:
       allOf:
```

Comment:
Do we still need `allOf` in case of `union`?

Reply:
You are right, with union, `allOf` is not required. I removed it.

Reply:
LGTM. I will merge it soon. Hi @Fokko, do you want to take another look?

Reply:
Sorry for the late reply, I was traveling. Thanks @anuragmantri and @flyrain for fixing this 👍